src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,302 @@
+            System.out.println(HORIZONTAL_LINE);
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(INDENT + taskArrayList.get(doneTaskNum).taskNum + "." + taskArrayList.get(doneTaskNum));
+        } else {
+            throw new NumberOutOfBoundsException();
+        }
+    }
+
+    private static void performDelete(int inputNum) throws NumberOutOfBoundsException {
+        //index of taskNum in taskArrayList
+        int deleteTaskNum = inputNum - 1;
+
+        boolean isValidNum = (inputNum > 0) && (inputNum <= Task.taskCount);
+        if (isValidNum) {
+            System.out.println(HORIZONTAL_LINE);
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(INDENT + taskArrayList.get(deleteTaskNum).taskNum + "." + taskArrayList.get(deleteTaskNum));
+            taskArrayList.remove(deleteTaskNum);
+            updateTaskCount();
+            updateTaskNum(deleteTaskNum);
+            System.out.println("Now you have " + Task.taskCount + " tasks in the list");
+        } else {
+            throw new NumberOutOfBoundsException();
+        }
+    }
+
+    private static void updateTaskCount(){
+        Task.taskCount = Task.taskCount - 1;
+    }
+
+    private static void updateTaskNum(int deleteTaskNum){
+        for (int i = deleteTaskNum; i < taskArrayList.size(); i++){
+            taskArrayList.get(i).taskNum = taskArrayList.get(i).taskNum - 1;
+        }
+    }
+
+    private static void splitTodo(String userInput) throws StringIndexOutOfBoundsException {
+        todoTask = userInput.replace("todo", "");
+        todoTask = todoTask.trim();
+        if (todoTask.equals("")) {
+            throw new StringIndexOutOfBoundsException();
+        }
+    }
+
+    private static void splitDeadline(String userInput) throws StringIndexOutOfBoundsException,
+            ByEmptyException {
+        dividerPosition = userInput.indexOf("/by");
+        deadlineDescription = userInput.substring(START_OF_STRING, dividerPosition);
+        deadlineDescription = deadlineDescription.replace("deadline", "");
+        deadlineDescription = deadlineDescription.trim();
+        charAfterDividerPosition = dividerPosition + 1;
+        by = userInput.substring(charAfterDividerPosition);
+        by = by.replace("by", "");
+        by = by.trim();
+        if (by.equals("")) {
+            throw new ByEmptyException();
+        }
+    }
+
+    private static void splitEvent(String userInput) throws StringIndexOutOfBoundsException,
+            AtEmptyException {
+        dividerPosition = userInput.indexOf("/at");
+        eventDescription = userInput.substring(START_OF_STRING, dividerPosition);
+        eventDescription = eventDescription.replace("event", "");
+        eventDescription = eventDescription.trim();
+        charAfterDividerPosition = dividerPosition + 1;
+        at = userInput.substring(charAfterDividerPosition);
+        at = at.replace("at", "");
```

Comment:
You may want to use a more descriptive variable name to improve code readability.